extensions/apicurio-registry-avro/deployment/src/main/java/io/quarkus/apicurio/registry/binding/ServiceRegistryBindingExtensionProcessor.java

```diff
@@ -0,0 +1,21 @@
+package io.quarkus.apicurio.registry.avro.binding;
+
+import io.quarkus.apicurio.registry.binding.ServiceRegistryBindingConverter;
+import io.quarkus.deployment.Capabilities;
+import io.quarkus.deployment.Capability;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
+
+class ServiceRegistryBindingExtensionProcessor {
+
+    @BuildStep
+    void registerServiceBinding(Capabilities capabilities,
+            BuildProducer<ServiceProviderBuildItem> serviceProvider) {
+        if (capabilities.isPresent(Capability.KUBERNETES_SERVICE_BINDING)) {
+            serviceProvider.produce(
+                    new ServiceProviderBuildItem("io.quarkus.apicurio.registry.binding.ServiceRegistryBindingConverter",
```

Comment:
Not related to that PR, but maybe we should have an SPI in the service binding extension with a dedicated build item (which will also about the capability check)

Reply:
I am not sure what you mean here. Can you elaborate a bit please?